packages/next-auth/src/providers/shopify.js

```diff
@@ -0,0 +1,79 @@
+export default function Shopify(options) {
+  const {
+    shop = "",
+    scope = "",
+    clientId = "",
+    clientSecret = "",
+    apiVersion = "",
+  } = options
+
+  return {
+    id: "shopify",
+    name: "Shopify",
+    type: "oauth",
+    authorization: {
+      url: `https://${shop}.myshopify.com/admin/oauth/authorize`,
+      params: {
+        scope,
```

Comment:
Let's set some sensible default scopes that are only enough to show basic profile information. If the developer wants to set additional scopes, they can easily override these defaults.

Reply:
We can add default scopes related to the profile but if the Shopify app wants to access the shop orders and their customers then again the shop admin need to reinstall the app with new scopes and it breaks the user experience. This is the reason all Shopify apps are suggested to add all the required scopes at once while installation.